src/plugins/src/lib.rs

```diff
@@ -88,41 +118,49 @@ fn ty_snake_to_camel(cx: &mut ExtCtxt, sp: Span, tts: &[TokenTree]) -> Box<MacRe
     } else {
         unreachable!()
     }
-    MacEager::ty(P(Ty {id: ast::DUMMY_NODE_ID, node: ty, span: sp}))
+    MacEager::ty(P(Ty {
+        id: ast::DUMMY_NODE_ID,
+        node: ty,
+        span: sp,
+    }))
 }
 
-fn convert(ident: &mut Ident) {
+/// Converts an ident in-place to CamelCase and returns the previous ident.
+fn convert(ident: &mut Ident) -> String {
     let ident_str = ident.to_string();
     let mut camel_ty = String::new();
 
-    // Find the first non-underscore and add it capitalized.
-    let mut chars = ident_str.chars();
-
-    // Find the first non-underscore char, uppercase it, and append it.
-    // Guaranteed to succeed because all idents must have at least one non-underscore char.
-    camel_ty.extend(chars.find(|&c| c != '_').unwrap().to_uppercase());
-
-    // When we find an underscore, we remove it and capitalize the next char. To do this,
-    // we need to ensure the next char is not another underscore.
-    let mut chars = chars.coalesce(|c1, c2| {
-        if c1 == '_' && c2 == '_' {
-            Ok(c1)
-        } else {
-            Err((c1, c2))
-        }
-    });
-
-    while let Some(c) = chars.next() {
-        if c != '_' {
-            camel_ty.push(c);
-        } else {
-            if let Some(c) = chars.next() {
-                camel_ty.extend(c.to_uppercase());
+    {
```

Comment:
Unfortunately this is just a bad diff; there's no code change here, just indentation.